lib/phoenix_live_view/channel.ex

```diff
@@ -128,32 +131,51 @@ defmodule Phoenix.LiveView.Channel do
   end
 
   def handle_info(%Message{topic: topic, event: "progress"} = msg, %{topic: topic} = state) do
-    %{socket: socket} = state
-    %{"ref" => ref, "entry_ref" => entry_ref, "progress" => progress} = msg.payload
-    new_socket = Upload.update_progress(socket, ref, entry_ref, progress)
-    upload_conf = Upload.get_upload_by_ref!(new_socket, ref)
-    entry = UploadConfig.get_entry_by_ref(upload_conf, entry_ref)
-
-    if event = entry && upload_conf.progress_event do
-      {:noreply, new_socket} = event.(upload_conf.name, entry, new_socket)
-      handle_changed(state, new_socket, msg.ref)
-    else
-      handle_changed(state, new_socket, msg.ref)
-    end
+    cid = msg.payload["cid"]
+
+    {new_state, _} =
+      with_socket(state, cid, msg.ref, fn socket, _ ->
+        %{"ref" => ref, "entry_ref" => entry_ref, "progress" => progress} = msg.payload
+        new_socket = Upload.update_progress(socket, ref, entry_ref, progress)
+        upload_conf = Upload.get_upload_by_ref!(new_socket, ref)
+        entry = UploadConfig.get_entry_by_ref(upload_conf, entry_ref)
+
+        if event = entry && upload_conf.progress_event do
+          {:noreply, new_socket} = event.(upload_conf.name, entry, new_socket)
+          {new_socket, {:ok, nil}}
+        else
+          {new_socket, {:ok, nil}}
+        end
+      end)
+
+    {:noreply, new_state}
   end
 
   def handle_info(%Message{topic: topic, event: "allow_upload"} = msg, %{topic: topic} = state) do
-    %{"ref" => ref, "entries" => entries} = msg.payload
-    upload_conf = Upload.get_upload_by_ref!(state.socket, ref)
-
-    {_ok_or_error, reply, %Socket{} = new_socket} =
-      with {:ok, new_socket} <- Upload.put_entries(state.socket, upload_conf, entries) do
-        Upload.generate_preflight_response(new_socket, upload_conf.name)
-      end
+    %{"ref" => upload_ref, "entries" => entries} = payload = msg.payload
+    cid = payload["cid"]
+
+    {new_state, new_upload_names} =
+      with_socket(state, cid, msg.ref, fn socket, _ ->
+        socket = Upload.register_cid(socket, upload_ref, cid)
```

Comment:
Maybe it's a simple naming thing, but does `register_cid` make sense as a function name when `cid` can be `nil`?